packages/webpack/src/BuildMetadataPlugin.ts

```diff
@@ -0,0 +1,118 @@
+import { execSync } from 'node:child_process';
+import * as path from 'node:path';
+import { PackageJson } from 'type-fest';
+import { type WebpackPluginInstance, BannerPlugin, Compilation, Compiler, sources } from 'webpack';
+
+const getBaseBuildMetadata = () => {
```

Comment:
can we have one get base build Metadata method in the config dir and use it for rollup and webpack ?

Reply:
It is more involved than expected, but it is doable.

Reply:
...see current code